From: Gerald Schaefer <geraldsc@de.ibm.com>
Subject: dasd: let recovery cqr inherit flags from failed cqr
References: bnc#631075,LTC#66750
Patch-mainline: Yes

Symptom:     Recovery requests behave not as expected, e.g. failfast
             is not working.
Problem:     The usual way to recover a failed DASD ECKD request (cqr)
             is to create a new request with an appropriate recovery
             CCW program. Certain features, e.g. failfast, can be
             enabled per request and are stored in the requests flags.
             The problem is, that these flags were not copied from the
             failed to the recovery request.
Solution:    Copy cqr flags when creating a recovery cqr.

Acked-by: John Jolly <jjolly@suse.de>
---
 drivers/s390/block/dasd_3990_erp.c |    3 +++
 1 file changed, 3 insertions(+)

--- a/drivers/s390/block/dasd_3990_erp.c
+++ b/drivers/s390/block/dasd_3990_erp.c
@@ -221,6 +221,7 @@ dasd_3990_erp_DCTL(struct dasd_ccw_req *
 	ccw->cmd_code = CCW_CMD_DCTL;
 	ccw->count = 4;
 	ccw->cda = (__u32)(addr_t) DCTL_data;
+	dctl_cqr->flags = erp->flags;
 	dctl_cqr->function = dasd_3990_erp_DCTL;
 	dctl_cqr->refers = erp;
 	dctl_cqr->startdev = device;
@@ -1674,6 +1675,7 @@ dasd_3990_erp_action_1B_32(struct dasd_c
 	ccw->cda = cpa;
 
 	/* fill erp related fields */
+	erp->flags = default_erp->flags;
 	erp->function = dasd_3990_erp_action_1B_32;
 	erp->refers = default_erp->refers;
 	erp->startdev = device;
@@ -2314,6 +2316,7 @@ static struct dasd_ccw_req *dasd_3990_er
 		ccw->cda      = (long)(cqr->cpaddr);
 	}
 
+	erp->flags = cqr->flags;
 	erp->function = dasd_3990_erp_add_erp;
 	erp->refers   = cqr;
 	erp->startdev = device;
